ui/pages/confirmations/hooks/useCurrentSignatureSecurityAlertResponse.ts

```diff
@@ -0,0 +1,40 @@
+import { useSelector } from 'react-redux';
+import { TransactionMeta } from '@metamask/transaction-controller';
+
+import { SecurityAlertResponse } from '../types/confirm';
+import { useConfirmContext } from '../context/confirm';
+
+type SecurityAlertResponsesState = {
+  metamask: {
+    signatureSecurityAlertResponses: Record<string, SecurityAlertResponse>;
+    transactions: TransactionMeta[];
+  };
+};
+
+const useCurrentSignatureSecurityAlertResponse = ():
+  | SecurityAlertResponse
+  | undefined => {
+  const { currentConfirmation } = useConfirmContext();
+
+  const securityAlertId = (
+    currentConfirmation?.securityAlertResponse as SecurityAlertResponse
+  )?.securityAlertId as string;
+
+  const signatureSecurityAlertResponse = useSelector(
+    (state: SecurityAlertResponsesState) => {
+      const signatureSecurityAlertId = (
+        currentConfirmation?.securityAlertResponse as SecurityAlertResponse
+      )?.securityAlertId;
+      if (securityAlertId === undefined) {
+        return undefined;
+      }
```

Comment:
did we want to check signatureSecurityAlertId here?
```suggestion
      if (signatureSecurityAlertId === undefined) {
        return undefined;
      }
```

Reply:
Done